torchvision/datasets/caltech.py

```diff
@@ -26,17 +26,15 @@ class Caltech101(VisionDataset):
             downloaded again.
     """
 
-    def __init__(self, root, target_type="category",
-                 transform=None, target_transform=None,
-                 download=False):
-        super(Caltech101, self).__init__(os.path.join(root, 'caltech101'))
+    def __init__(self, root, target_type="category", transforms=None, transform=None,
```

Comment:
Even though `transforms` is the new "recommended" way, changing the order of the arguments is a BC-breaking change and can lead to subtle bugs for the user.

Also, I'm unclear of the benefits of adding the `transforms` for classification-only datasets, as we generally do not need to perform joint transforms for the images and the targets.

Reply:
> Even though `transforms` is the new "recommended" way, changing the order of the arguments is a BC-breaking change and can lead to subtle bugs for the user.

I didn't think of this.

> Also, I'm unclear of the benefits of adding the transforms for classification-only datasets, as we generally do not need to perform joint transforms for the images and the targets.

It was my understanding that the joint `transforms`, the new way, is or will be the _only_ way to go in the future (hence the above question about deprecating the "old" way). Is this not intended for classification datasets?

Reply:
The joint `transforms` is the new way, but we shouldn't force it to be the only way for classification datasets because it will not fit nicely with the `transforms` that we have, which assume only single images as input.

So I think that the added benefit of having a more general `transforms` for classification datasets doesn't pay off as of now, because our story around multi-type transforms is not very well defined yet.

Can you leave the current functionality as is, and we can discuss on a per dataset basis of what other datasets should support `transforms`?

Reply:
So remove the `transforms` argument again and only keep the call of `VisionDataset` constructor with `transform` and `target_transform`?

Reply:
Yes, precisely! I think this is better for now

Reply:
Since the constructor of `VisionDataset` has `transforms` as first argument

https://github.com/pytorch/vision/blob/bbd363ca2713fb68e1e190206578e600a87baf90/torchvision/datasets/vision.py#L9

do you want me to do 

```python
super(Dataset, self).__init__(root, transform=transform, target_transform=target_transform)
```

or change the order so that we can call it with positional arguments?

```python
super(Dataset, self).__init__(root, transform, target_transform)
```

Reply:
Passing it as named arguments is what I would do, to avoid modifying `VisionDataset`

Reply:
Fixed in 0df7caa